crates/starknet-devnet/src/main.rs

```diff
@@ -198,20 +204,51 @@ async fn main() -> Result<(), anyhow::Error> {
 
     info!("Starknet Devnet listening on {}", address);
 
-    if starknet_config.dump_on == Some(DumpOn::Exit) {
-        server.with_graceful_shutdown(shutdown_signal(api.clone())).await?
-    } else {
-        server.await?
+    let empty_handle: task::JoinHandle<()> = task::spawn(async {});
+    let mut block_interval_handle = empty_handle;
+    if let BlockGeneration::Interval(timestamp) = starknet_config.block_generation_on {
+        // use JoinHandle to run block interval creation as a task
+        block_interval_handle = task::spawn(create_block_interval(api.clone(), timestamp));
     }
 
+    // run server also as a JoinHandle
+    let server_handle =
+        task::spawn(server.with_graceful_shutdown(shutdown_signal(api.clone())).into_future());
+
+    // wait for ctrl + c signal (SIGINT)
+    shutdown_signal(api.clone()).await;
+
+    // join both tasks
+    let _ = tokio::join!(block_interval_handle, server_handle);
+
     Ok(())
 }
 
+async fn create_block_interval(api: Api, block_interval: u64) {
+    let mut interval = interval(Duration::from_secs(block_interval));
+    let mut sigint = signal(SignalKind::interrupt()).expect("Failed to setup SIGINT handler");
+
+    loop {
+        tokio::select! {
+            _ = interval.tick() => {
+                let mut starknet = api.starknet.write().await;
+                let _ = starknet.create_block_dump_event(None);
+            }
+            _ = sigint.recv() => {
+                return;
+            }
+        }
+    }
+}
+
 pub async fn shutdown_signal(api: Api) {
     tokio::signal::ctrl_c().await.expect("Failed to install CTRL+C signal handler");
```

Comment:
I remember there were some problems with doing Ctrl+C if running Devnet as a Docker container (had to install `tini` to take care of it). Can you try building a Docker image, running it in differen block generation modes and seeing if they are all exitable by means of a single Ctrl+C?

Building is achievable by running this in the repository root:
```
$ docker build -t shardlabs/starknet-devnet-rs:dummy -f docker/Dockerfile .
```
Then running is
```
$ docker run --rm -it -p 127.0.0.1:5050:5050 shardlabs/starknet-devnet-rs:dummy
```

Reply:
so I tested it like this `starknet-devnet --block-generation-on 1 --dump-on exit --dump-path test2 --port 123` and there was something in test2 after some time.

```
# cat test2
["CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock","CreateBlock"]# 
```

Reply:
but I spotted another problem there were no info prints

Reply:
Fixed in e07f02f403a8b0737fc3ff141a542361940c2b82

Reply:
ok, assuming exiting dockerized Devnet is still possible with just one Ctrl+C, we can resolve this issue

Reply:
yes it's working with just one ctrl + c